x-pack/plugins/infra/public/components/infrastructure_node_metrics_tables/shared/hooks/metrics_to_api_options.test.ts

```diff
@@ -5,44 +5,59 @@
  * 2.0.
  */
 
-import type { MetricsMap } from './metrics_to_api_options';
+import type { MetricsExplorerOptions } from '../../../../pages/metrics/metrics_explorer/hooks/use_metrics_explorer_options';
+import { createMetricByFieldLookup, MetricsQueryOptions } from './metrics_to_api_options';
 import { metricsToApiOptions } from './metrics_to_api_options';
 
 describe('metricsToApiOptions', () => {
   type TestNodeTypeMetricsField = 'test.node.type.field1' | 'test.node.type.field2';
 
-  const testMetricsMapField1First: MetricsMap<TestNodeTypeMetricsField> = {
-    'test.node.type.field1': {
-      aggregation: 'max',
-      field: 'test.node.type.field1',
+  const testMetricsMapField1First: MetricsQueryOptions<TestNodeTypeMetricsField> = {
+    sourceFilter: {
+      term: {
+        'event.module': 'test',
+      },
     },
-    'test.node.type.field2': {
-      aggregation: 'avg',
-      field: 'test.node.type.field2',
+    groupByField: 'test.node.type.groupingField',
+    metricsMap: {
+      'test.node.type.field1': {
+        aggregation: 'max',
+        field: 'test.node.type.field1',
+      },
+      'test.node.type.field2': {
+        aggregation: 'avg',
+        field: 'test.node.type.field2',
+      },
     },
   };
 
-  const testMetricsMapField1Second: MetricsMap<TestNodeTypeMetricsField> = {
-    'test.node.type.field2': {
-      aggregation: 'avg',
-      field: 'test.node.type.field2',
+  const testMetricsMapField1Second: MetricsQueryOptions<TestNodeTypeMetricsField> = {
+    sourceFilter: {
+      term: {
+        'event.module': 'test',
+      },
     },
-    'test.node.type.field1': {
-      aggregation: 'max',
-      field: 'test.node.type.field1',
+    groupByField: 'test.node.type.groupingField',
+    metricsMap: {
+      'test.node.type.field2': {
+        aggregation: 'avg',
+        field: 'test.node.type.field2',
+      },
+      'test.node.type.field1': {
+        aggregation: 'max',
+        field: 'test.node.type.field1',
+      },
     },
   };
 
   const fields = ['test.node.type.field1', 'test.node.type.field2'];
 
   it('should join the grouping field with the metrics in the APIs expected format', () => {
-    const { options } = metricsToApiOptions(
-      testMetricsMapField1First,
-      'test.node.type.groupingField'
-    );
+    const { options } = metricsToApiOptions(testMetricsMapField1First);
     expect(options).toEqual({
       aggregation: 'avg',
       groupBy: 'test.node.type.groupingField',
+      filterQuery: JSON.stringify({ bool: { filter: [{ term: { 'event.module': 'test' } }, {}] } }),
```

Comment:
We don't want this empty object in here, right?

Reply:
yup. it's outdated